File: source_funcs/go_transcoder/FIND_PAIR_WITH_GREATEST_PRODUCT_IN_ARRAY/FIND_PAIR_WITH_GREATEST_PRODUCT_IN_ARRAY.go
package main

func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}

func f_gold(arr []int, n int) int {
	var result int = -1
	for i := int(0); i < n; i++ {
		for j := int(0); j < n-1; j++ {
			for k := int(j + 1); k < n; k++ {
				if arr[j]*arr[k] == arr[i] {
					result = max(result, arr[i])
				}
			}
		}
	}
	return result
}

File: rust_funcs/go_transcoder/FIND_PAIR_WITH_GREATEST_PRODUCT_IN_ARRAY/FIND_PAIR_WITH_GREATEST_PRODUCT_IN_ARRAY.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}


fn f_gold(arr: [i32;2], n: i32) -> i32 {  
let mut result = -1;
for i in 0..n as usize {     
  for j in 0..n as usize - 1 {     
    for k in j+1..n as usize {
      if arr[j] as f32 * arr[k] as f32 == arr[i] as f32 { 
         result = result.max(arr[i]);  
      }
    }
  }
}
return result;
}
